drivers/hopwa_caper/app/models/hopwa_caper/generators/fy2024/generator.rb

```diff
@@ -0,0 +1,162 @@
+      return question_number
+    end
+
+    def self.filter_class
+      ::HopwaCaper::Filters::HopwaCaperFilter
+    end
+
+    def self.allowed_options
+      [
+        :start,
+        :end,
+        :coc_codes,
+        :project_ids,
+        :data_source_ids,
+        :project_type_codes,
+        :project_group_ids,
+      ]
+    end
+
+    protected
+
+    def service_history_enrollments
+      scope = GrdaWarehouse::ServiceHistoryEnrollment.
+        entry.
+        open_between(start_date: @report.start_date, end_date: @report.end_date)
+      # FIXME- I don't know how to create these from a spec
+      return scope if Rails.env.test?
+
+      @filter = self.class.filter_class.new(
+        user_id: @report.user_id,
+        enforce_one_year_range: false,
+      ).update(@report.options)
+
+      @filter.project_ids = @report.project_ids
+      @filter.apply(scope)
+    end
+
+    def build_hopwa_caper_models
+      scope = service_history_enrollments.preload(enrollment: [:income_benefits, :client, :disabilities, { project: :funders }])
+      scope.in_batches(of: 100, order: :desc) do |batch|
+        enrollment_rows = []
+        service_rows = []
+        batch.each do |service_history_enrollment|
+          hud_enrollment = service_history_enrollment.enrollment
+
+          enrollment_rows << HopwaCaper::Enrollment.from_hud_record(report: report, enrollment: hud_enrollment)
+          service_rows += hud_enrollment.services.map do |hud_service|
+            HopwaCaper::Service.from_hud_record(report: report, enrollment: hud_enrollment, service: hud_service)
+          end
+        end
+
+        populate_hopwa_eligible(enrollment_rows)
+        ensure_uniform_client_attrs(enrollment_rows)
+
+        import_rows(HopwaCaper::Enrollment, enrollment_rows)
+        import_rows(HopwaCaper::Service, service_rows)
+      end
+    end
+
+    # ensure consistent values for individuals (can vary based on enrollment entry date)
+    def ensure_uniform_client_attrs(enrollment_rows)
+      groups = enrollment_rows.sort_by(&:id).group_by { |e| [e.data_source_id, e.hud_personal_id] }.values
+      groups.each do |group|
+        uniform_attrs = {
+          age: group.map(&:age).compact.max,
+          hiv_positive: groups.any?(:hiv_positive),
+          ever_perscribed_anti_retroviral_therapy: groups.any?(:ever_perscribed_anti_retroviral_therapy),
+          viral_load_supression: groups.any?(:viral_load_supression),
+        }
+        group.each { |e| e.attributes = uniform_attrs }
+      end
+    end
+
+    # try and figure out which person in a household is hopwa eligible
+    def populate_hopwa_eligible(enrollment_rows)
+      households = enrollment_rows.sort_by(&:id).group_by { |e| [e.data_source_id, e.hud_household_id] }.values
+      eligible_enrollments = households.map do |enrollments|
+        # if the hoh is hiv+
+        hohs = enrollments.filter { |e| e.relationship_to_hoh == 1 }
```

Comment:
Might as well lean on existing helper methods.
```suggestion
        hohs = enrollments.filter(&:head_of_household?)
```

Reply:
enrollments here are hopwa cper enrollments, so the helper isn't available